cmd/update-main.go

```diff
@@ -68,90 +68,112 @@ const (
 	minioReleaseURL           = "https://dl.minio.io/server/minio/release/" + runtime.GOOS + "-" + runtime.GOARCH + "/"
 )
 
-func getCurrentReleaseTime(minioVersion, minioBinaryPath string) (releaseTime time.Time, err error) {
-	if releaseTime, err = time.Parse(time.RFC3339, minioVersion); err == nil {
+type updateParams struct {
+	versionStr         string
+	serverBinaryPath   string
+	cgroupFile         string
+	releaseChecksumURL string
+	serverMode         string
+}
+
+// newUpdateParams - returns appropriate default value for
+// updateParams
+func newUpdateParams() *updateParams {
+	return &updateParams{
+		versionStr:         Version,
+		serverBinaryPath:   os.Args[0],
+		cgroupFile:         "/proc/self/cgroup",
+		releaseChecksumURL: minioReleaseURL + "minio.shasum",
+		serverMode:         "",
+	}
+}
+
+// getCurrentReleaseTime - returns this process's release time.  If it
+// is official minio version, parsed version is returned else minio
+// binary's mod time is returned.
+func (u *updateParams) getCurrentReleaseTime() (releaseTime time.Time,
+	err error) {
+
+	if releaseTime, err = time.Parse(time.RFC3339, u.versionStr); err == nil {
 		return releaseTime, err
 	}
 
-	if !filepath.IsAbs(minioBinaryPath) {
-		// Make sure to look for the absolute path of the binary.
-		minioBinaryPath, err = exec.LookPath(minioBinaryPath)
+	if !filepath.IsAbs(u.serverBinaryPath) {
+		// Make sure to look for the absolute path of the
+		// binary.
+		u.serverBinaryPath, err = exec.LookPath(u.serverBinaryPath)
 		if err != nil {
 			return releaseTime, err
 		}
 	}
 
 	// Looks like version is minio non-standard, we use minio binary's ModTime as release time.
-	fi, err := osStat(minioBinaryPath)
+	fi, err := osStat(u.serverBinaryPath)
 	if err != nil {
-		err = fmt.Errorf("Unable to get ModTime of %s. %s", minioBinaryPath, err)
+		err = fmt.Errorf("Unable to get ModTime of %s. %s",
+			u.serverBinaryPath, err)
 	} else {
 		releaseTime = fi.ModTime().UTC()
 	}
 
 	return releaseTime, err
 }
 
-// GetCurrentReleaseTime - returns this process's release time.  If it is official minio version,
-// parsed version is returned else minio binary's mod time is returned.
-func GetCurrentReleaseTime() (releaseTime time.Time, err error) {
-	return getCurrentReleaseTime(Version, os.Args[0])
-}
-
-func isDocker(cgroupFile string) (bool, error) {
-	cgroup, err := ioutil.ReadFile(cgroupFile)
+func (u *updateParams) isDockerInternal() (bool, error) {
+	cgroup, err := ioutil.ReadFile(u.cgroupFile)
 	if os.IsNotExist(err) {
 		err = nil
 	}
 
 	return bytes.Contains(cgroup, []byte("docker")), err
 }
 
-// IsDocker - returns if the environment is docker or not.
-func IsDocker() bool {
-	found, err := isDocker("/proc/self/cgroup")
+// isDocker - returns if the environment is docker or not.
+func (u *updateParams) isDocker() bool {
+	found, err := u.isDockerInternal()
 	fatalIf(err, "Error in docker check.")
```

Comment:
Improved it - we dont have to make it specific to the cgroup file as the check may change in the future.